src/main/java/org/fenixedu/academicextensions/services/registrationhistory/RegistrationHistoryReport.java

```diff
@@ -792,6 +794,15 @@ public String getQualificationDegreeDesignation() {
         return info == null ? null : info.getDegreeDesignation();
     }
 
+    public String getQualificationCountry() {
+        final PrecedentDegreeInformation info = getCompletedPrecedentInformation();
```

Comment:
Since this class ir already big, we could try to reduce the number of new lines of code.. Maybe using Optionals here could help?